src/main/java/pl/javastart/task/services/MixPhoneContract.java

```diff
@@ -0,0 +1,76 @@
+package pl.javastart.task.services;
+
+public class MixPhoneContract extends CardPhoneContract {
+    int remainingSms;
+    int remainingMms;
+    double remainingMinutes;
+
+    public MixPhoneContract(double balance, double smsFee, double mmsFee, double minuteCallFee,
+                            int smsAvailable, int mmsAvailable, double minutesAvailable) {
+        super(balance, smsFee, mmsFee, minuteCallFee);
+        this.remainingSms = smsAvailable;
+        this.remainingMms = mmsAvailable;
+        this.remainingMinutes = minutesAvailable;
+    }
+
+    @Override
+    int availableCallSeconds(int seconds) {
+        double remainingMinutesAsSeconds = (int) (remainingMinutes * 60);
+        double allowanceSeconds = 0;
+        if (remainingMinutesAsSeconds >= seconds) { // pobiera wszystko z darmowych minut
+            callInSeconds = callInSeconds + seconds;
+            allowanceSeconds = seconds;
+            remainingMinutes -= allowanceSeconds / 60;
+            return (int) allowanceSeconds;
+        } else if (remainingMinutesAsSeconds > 0 && remainingMinutesAsSeconds < seconds) { // wybiera reszte z darmowych minut
+            remainingMinutes = 0;
+            callInSeconds = (int) (callInSeconds + remainingMinutesAsSeconds);
+            allowanceSeconds += remainingMinutesAsSeconds;
+        }
+        int secondsHasToBePayedFromCredit = (int) (seconds - remainingMinutesAsSeconds);
+        allowanceSeconds += super.availableCallSeconds(secondsHasToBePayedFromCredit);
+        return (int) allowanceSeconds;
```

Comment:
Trochę ciężko się połapać co się tutaj dzieje, bo nazwa metody, czyli availableCallSeconds nie mówi o tym co będzie zwrócone (metody powinny być też czasownikami). Masz zmienną allowanceSeconds, czyli jak rozumiem "odliczanie sekund", ale nie wiadomo, czy to oznacza czas rozmowy (callTime?), czy pozostały czas rozmowy (callTimeRemaining?), czy może jeszcze coś innego. Bez głębokiego wczytywania się w kod ciężko to zrozumieć.

Po wczytaniu się w tę metodę wydaje się, że działa ona poprawnie, ale zdecydowanie czytelniej byłoby wydzielić jakieś metody, które nadałyby w tym kodzie znaczenie poszczególnym fragmentom. Coś w stylu:
```
int remainingSeconds = useMinutesPackage(seconds);
if (remainingSeconds > 0) {
    remainingSeconds = useAccountBalance(seconds);
}
return seconds - remainingSeconds; //zakładając, że zwracamy czas rozmowy
```
Jeżeli dasz radę tutaj to upiększyć, to myślę, że warto jeszcze spróbować, ale jeżeli zostanie tak, jak jest no to też ok, bo sam algorytm wygląda w porządku i nie będę za to ucinał punktów.